registration/include/pcl/registration/correspondence_estimation_lookup_table.h

```diff
@@ -0,0 +1,517 @@
+    };
+
+    /** \brief @b CorrespondenceEstimationLookupTable provides fast correspondence estimation
+      * by using a lookup table with pre-computed correspondences.
+      *
+      * \author Carlos M. Costa
+      * \ingroup registration
+      */
+    template <typename PointSource, typename PointTarget, typename Scalar = float>
+    class CorrespondenceEstimationLookupTable: public CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>
+    {
+      public:
+        typedef boost::shared_ptr<CorrespondenceEstimationLookupTable<PointSource, PointTarget, Scalar> > Ptr;
+        typedef boost::shared_ptr<const CorrespondenceEstimationLookupTable<PointSource, PointTarget, Scalar> > ConstPtr;
+
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::corr_name_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::force_no_recompute_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::force_no_recompute_reciprocal_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::indices_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::initCompute;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::initComputeReciprocal;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::input_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::target_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::tree_;
+        using CorrespondenceEstimationBase<PointSource, PointTarget, Scalar>::tree_reciprocal_;
+        using PCLBase<PointSource>::deinitCompute;
+
+        typedef pcl::search::KdTree<PointTarget> KdTree;
+        typedef typename pcl::search::KdTree<PointTarget>::Ptr KdTreePtr;
+
+        typedef pcl::search::KdTree<PointSource> KdTreeReciprocal;
+        typedef typename KdTree::Ptr KdTreeReciprocalPtr;
+
+        typedef pcl::PointCloud<PointSource> PointCloudSource;
+        typedef typename PointCloudSource::Ptr PointCloudSourcePtr;
+        typedef typename PointCloudSource::ConstPtr PointCloudSourceConstPtr;
+
+        typedef pcl::PointCloud<PointTarget> PointCloudTarget;
+        typedef typename PointCloudTarget::Ptr PointCloudTargetPtr;
+        typedef typename PointCloudTarget::ConstPtr PointCloudTargetConstPtr;
+
+        typedef typename KdTree::PointRepresentationConstPtr PointRepresentationConstPtr;
+
+        /** \brief Empty constructor. */
+        CorrespondenceEstimationLookupTable ()
+        {
+          corr_name_  = "CorrespondenceEstimationLookupTable";
+        }
+
+        /** \brief Empty destructor. */
+        virtual ~CorrespondenceEstimationLookupTable () {}
+
+        /** \brief Get the source lookup table. */
+        inline CorrespondenceLookupTable<PointSource>&
+        getSourceCorrespondencesLookupTable () { return (source_correspondences_lookup_table_); }
```

Comment:
Do you really want to allow this to be changed from outside the class? If yes, you need to provide a const alternative.